82/step3.cpp

```diff
@@ -0,0 +1,19 @@
+class Solution {
+public:
+    ListNode* deleteDuplicates(ListNode* head) {
+        ListNode dummy_head = ListNode(-1, head);
+        ListNode* current = &dummy_head;
+        while (current) {
+            ListNode* group_top = current->next;
+            while (group_top && group_top->next && group_top->val == group_top->next->val) {
+                int group_val  =group_top->val;
+                while (group_top && group_top->val == group_val) {
+                    group_top = group_top->next;
+                }
+            }
+            current->next = group_top;
+            current = current->next;
+        }
```

Comment:
while 3重が気になりました
if 文を挟めば2重にできるかなと思います
```suggestion
while (current) {
            ListNode* group_top = current->next;
            if (group_top && group_top->next && group_top->val == group_top->next->val)  {
                int group_val = group_top->val;
                while (group_top && group_top->val == group_val) {
                    group_top = group_top->next;
                }
                current->next = group_top;
            } else {
                current = current->next;
            }
            
        }
```

Reply:
3重whieは確かに微妙ですね・・・ifで別途書き直してみます。